### PR TITLE
ALP 105 Province drop down component in form

### LINKED ISSUE #105: The "Province" input of the VolunteerSignUpForm should be a drop down input
As per the specification: 
Screen Shot 2020-03-28 at 12 49 14 PM

We should be using the SelectBox component

### PR DESCRIPTION
**Pull request by michaelfromyeg**
Added province dropdown component to form, and I also noticed ID wasn't being return on get volunteers, so I fixed that quickly to stop the React error (child needs key).

I had to modify the SelectBox a fair bit to get this working, so I had to tweak the sign in page as well, so that I didn't have to create a third select component specifically for Formik.

### CODE REVIEW

File: aunt-leahs-functions/location/index.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ module.exports = function(context, req) {
 	connection.on('connect', (error) => {
 		if (error) {
 			context.log('Error: ', error);
-			context.done();
+			context.done(error);
```
Comment (justinkkwan) [inline aunt-leahs-functions/location/index.js]: Maybe rename `error` to `err` to be consistent with the other files

File: aunt-leahs-functions/shifts/index.js [hunk 2]
```diff
@@ -112,7 +112,7 @@ module.exports = function (context, req) {
 
         request.on('doneProc', function (rowCount, more, rows) {
             context.res = {
-                body: { message: 'Successfully inserted 1 row into the database' }
+                body: { message: 'Successfully inserted 1 row into the database'}
```
Comment (justinkkwan) [inline aunt-leahs-functions/shifts/index.js]: I feel like the client won't care about this message. I think we should return the actual message from the database instead. Or maybe instead just do  `success: true`?

**Review by justinkkwan — APPROVED** [review]
✔️ Seems good. Not sure about changing all the single quote to doubles though. It seems like most other files in the project are using single quotes

File: aunt-leahs-functions/location/index.js (hunk 1 above)
Comment (thaingocnguyen) [reply aunt-leahs-functions/location/index.js]: We should settle on a convention because I see that Justin does the opposite (change " to ') in his PR

**Review by thaingocnguyen — CHANGES_REQUESTED** [review]
Similar to TextInput, SelectBox needs helperText for the form validation feedback to show

File: aunt-leahs-app/src/pages/VolunteerCheckInPage.jsx [hunk 3]
```diff
@@ -11,23 +11,28 @@ import Header from '../components/Header'
...
+  const [duration, setDuration] = useState("");
 	const [isFormSubmitted, setIsFormSubmitted] = useState(false);
 	const [isSubmissionSuccessful, setIsSubmissionSuccessful] = useState(false);
 
 	const postShift = async () => {
 		try {
-
+      if (!duration.target) {
+        alert("Please select a shift time");
+        throw new Error("No time selected");
```
Comment (thaingocnguyen) [inline aunt-leahs-app/src/pages/VolunteerCheckInPage.jsx]: I think just the alert message is sufficient, I can't think of a reason why there should be a console error

File: aunt-leahs-functions/location/index.js (hunk 1 above)
Comment (michaelfromyeg) [reply aunt-leahs-functions/location/index.js]: Should be single quotes, I just forgot the changes my jsx Prettier settings on my desktop (I switched machines when I got home). Fixed my settings, and I'll update this PR accordingly!

**Review by codygagnon — APPROVED** [review]

**Review by thaingocnguyen — APPROVED** [review]
💯

**Pull request merged**